### PR TITLE
feat(api): added deleteaccount resolver

### PR DESCRIPTION
**Pull request by BronzW**

### CODE REVIEW

File: packages/api/.env.example [hunk 1]
```diff
@@ -11,7 +11,7 @@ OASIS_API_GITHUB_CALLBACK_URL=
 
 # https://apps.twitter.com
 OASIS_API_TWITTER_API_KEY=
-OASIS_API_TWITTER_API_SECRET_KEY=
```
Comment (HenryLeC) [inline packages/api/.env.example:14]: This is unrelated, can you please make a new PR for this. It is a lot of work to update env vars on the server lol

Comment (heybereket) [reply packages/api/.env.example:14]: > This is unrelated, can you please make a new PR for this. It is a lot of work to update env vars on the server lol

It should be without _API_ the other auth providers don't have it, someone must have added it by mistake.

**Review by Angshu31 — APPROVED** [review]

**Review by heybereket — APPROVED** [review]

File: packages/api/.env.example (hunk 1 above)
Comment (BronzW) [reply packages/api/.env.example:14]: So what I changed it to or what it was?

> > This is unrelated, can you please make a new PR for this. It is a lot of work to update env vars on the server lol
> 
> It should be without _API_ the other auth providers don't have it, someone must have added it by mistake.

File: packages/api/src/modules/user/DeleteAccount.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { ContextType } from '../../apolloServer';
+import { createConnection } from 'typeorm';
+
+@Resolver()
+export default class DeleteAccountResolver {
+  @Mutation(() => Boolean)
+  @Authorized()
+  async deleteAccount(@Ctx() { getUser }: ContextType) {
+    const user = await getUser();
+    await User.createQueryBuilder()
```
Comment (HenryLeC) [inline packages/api/src/modules/user/DeleteAccount.resolver.ts]: Don't use query builder use User.remove()

**Review by HenryLeC — CHANGES_REQUESTED** [review]

**Review by HenryLeC — APPROVED** [review]
Looks good ty

**Comment by BronzW** [comment]
Since all code owners approved this I will merge it
